core/local/local_test.go

```diff
@@ -1387,3 +1388,281 @@ func TestNewExecutionSchedulerHasWork(t *testing.T) {
 	assert.Len(t, execScheduler.executors, 2)
 	assert.Len(t, execScheduler.executorConfigs, 3)
 }
+
+func TestExecutionStatsVUSharing(t *testing.T) {
+	t.Parallel()
+	script := []byte(`
+		import exec from 'k6/execution';
+		import { sleep } from 'k6';
+
+		// The cvus scenario should reuse the two VUs created for the carr scenario.
+		export let options = {
+			scenarios: {
+				carr: {
+					executor: 'constant-arrival-rate',
+					exec: 'carr',
+					rate: 9,
+					timeUnit: '0.95s',
+					duration: '1s',
+					preAllocatedVUs: 2,
+					maxVUs: 10,
+					gracefulStop: '100ms',
+				},
+			    cvus: {
+					executor: 'constant-vus',
+					exec: 'cvus',
+					vus: 2,
+					duration: '1s',
+					startTime: '2s',
+					gracefulStop: '0s',
+			    },
+		    },
+		};
+
+		export function cvus() {
+			const stats = Object.assign({scenario: 'cvus', globalVUID: __VU}, exec.getVUStats());
```

Comment:
by globalVUID do you mean "global for the instance" or for the test (in case of execution segments usage). As it is definitely not the latter and making it will be a breaking change I don't think is worth it.

On the other hand we have no way to get the latter ("global VUI id for the whole test") which is something that is a particular problem for cloud tests.

We can probably add it similar to how global Iteration was added, but maybe in a separate PR?

Reply:
Maybe "global" was the wrong name for it, but it's `__VU`, so global for the test and instance, since no execution segments are used. It's just a label to ensure VUs are being reused in the test, but I can change it to something else if you want.

I don't think we discussed global VU IDs like you mention, so yeah, let's add it to a separate issue and PR.

Reply:
Should be fairly simple to add, so I'd prefer now if possible

Reply:
hmm actually it won't be all that simple, when you consider dynamically allocated VUs for the arrival-rate executors :disappointed: and if we currently don't calculate the total max VUs in every k6 instance, regardless of the segment, it might have performance implications if we just add this now... so probably a separate PR will be better :disappointed:

Reply:
> performance implications if we just add this now.

We do now calculate through segment indexes which number an iteration is globally ... so I doubt calculating that for a VU will be noticeable

Reply:
Tto calculate which number a VU is, you first need to get the total max number of possible VUs (i.e. without an execution segments), which may be a very expensive operation for some `ramping-vus` configs. Whereas the global iteration counter is just a segmented index with the current execution segment, for which we've already done the calculations :man_shrugging:

Reply:
why? 

as far as I am concern it is literally when you make a VU you have segmented index ... which gives it's local (unscaled) and global VU (scaled) number and that's it.

I don't care what is the max VUs it is the responsibility of the code making the VUs to make certain we don't go above that

Reply:
:man_facepalming: yeah, you are completely right, I had a stupid idea in my head about how this should be done, probably influenced by some cloud-related implementation details :sweat_smile: the simple segmented index will be perfectly adequate way to implement this :+1: